lib/StaticAnalyzer/Core/PlistDiagnostics.cpp

```diff
@@ -866,6 +879,8 @@ static std::string getMacroNameAndPrintExpansion(TokenPrinter &Printer,
     Printer.printToken(T);
   }
 
+  AlreadyProcessedTokens.erase(IDInfo);
```

Comment:
Wouldn't it be more performant and straightforward, to `AlreadyProcessedTokens.clear()` before starting the recursive call chain, at line 793 ?

Reply:
Or perhaps we just don't need to erase at all since `AlreadyProcessedTokens` is a local variable in `getExpandedMacro` ?